### PR TITLE
Closes #2528 Add bootstrap class `col-form-label` to `<legend>` element within `<fieldset>`s.

### LINKED ISSUE #2528: Better exposed filters checkbox option does not add appropriate bootstrap class to `<legend>` element.
## Problem/Motivation
I want my forms to use bootstrap classes as much as possible.

### Describe the bug
Better exposed filters is unaware of bootstrap, and needs to be modified to use bootstrap classes for the form elements it outputs.

### To Reproduce

Add a term or two to az_publication_category vocabulary
Add an exposed filter to  to allow filtering by category in the az_publication view
Allow multiple selection.
Use better exposed filters to convert to checkboxes.

See that the output contains an plain `<legend>` element.

The reboot.scss file included with bootstrap makes that text huge, and doesn't match the rest of the form.

Bootstrap also comes with a class that can be applied to `<legend>` elements to make it inherit the font size from the rest of the form. (ideal).

Example: https://digital.arizona.edu/arizona-bootstrap/docs/2.0/components/forms/#horizontal-form
```
  <fieldset class="form-group row">
  
    <legend class="col-form-label col-sm-2 float-sm-left pt-0">Radios</legend>
  
    <div class="col-sm-10">
      <div class="form-check">
        <input class="form-check-input" type="radio" name="gridRadios" id="gridRadios1" value="option1" checked>
        <label class="form-check-label" for="gridRadios1">
          First radio
        </label>
      </div>
      <div class="form-check">
        <input class="form-check-input" type="radio" name="gridRadios" id="gridRadios2" value="option2">
        <label class="form-check-label" for="gridRadios2">
          Second radio
        </label>
      </div>
      <div class="form-check disabled">
        <input class="form-check-input" type="radio" name="gridRadios" id="gridRadios3" value="option3" disabled>
        <label class="form-check-label" for="gridRadios3">
          Third disabled radio
        </label>
      </div>
    </div>
  </fieldset>
```

## Proposed resolution
Description of the proposed solution, the rationale behind it, and workarounds for people who cannot use the patch.

### Expected behavior

The checkbox label should be the same size as the other labels in the form.

image

image

### PR DESCRIPTION
**Pull request by trackleft**
## Description

Within the PR above you will see a new template for az_barrio that handles fieldsets.

Bootstrap's opinion on this: https://getbootstrap.com/docs/4.0/components/forms/#horizontal-form-label-sizing

The only change to the [existing template](https://git.drupalcode.org/project/bootstrap_barrio/-/blob/5.x/templates/form/fieldset.html.twig) is the addition of the `col-form-label` class to the  `legend` element.

```
{%
  set legend_classes = [
    'col-form-label',
   'font-weight-bold',
  ]
%}
```

Extensible
```
  <legend{{ legend.attributes.addClass(legend_classes) }}>
```

## Related issues
- Closes #2528 
- ~~Blocked by #3279 (blocked because it is difficult to test without demo content.)~~

## How to test

Go to [`/finders/events`](https://9a355b9d-e15c-41ca-a57e-f8b221c9a044.probo.build/finders/events)
Notice the "Category" fieldset legend in the exposed filter block in the left sidebar.
Compare with main branch: https://demo.quickstart.arizona.edu/finders/events

Go to [`/finders/people`](https://9a355b9d-e15c-41ca-a57e-f8b221c9a044.probo.build/finders/people)
Compare with main branch: https://demo.quickstart.arizona.edu/finders/people

Go to [`/finders/news`](https://9a355b9d-e15c-41ca-a57e-f8b221c9a044.probo.build/finders/news)
Compare with main branch: https://demo.quickstart.arizona.edu/finders/news

Go to [`/finders/pages`](https://9a355b9d-e15c-41ca-a57e-f8b221c9a044.probo.build/finders/pages)
Compare with main branch: https://demo.quickstart.arizona.edu/finders/pages

Go to [`/finders/content`](https://9a355b9d-e15c-41ca-a57e-f8b221c9a044.probo.build/finders/content)
Compare with main branch: https://demo.quickstart.arizona.edu/finders/content

It would probably also be worth testing on sites with web forms that have fieldsets with legends in them.

## Screenshots
image

### Arizona Quickstart (install profile, custom modules, custom theme)
- **Patch release changes**
   - [ ] Bug fix
   - [ ] Accessibility, performance, or security improvement
   - [ ] Critical institutional link or brand change
   - [ ] Adding experimental module
   - [ ] Update experimental module
- **Minor release changes**
   - [ ] New feature
   - [x] Breaking or visual change to existing behavior
   - [ ] Upgrade experimental module to stable
   - [ ] Enable existing module by default or database update
   - [ ] Non-critical brand change
   - [ ] New internal API or API improvement with backwards compatibility
   - [ ] Risky or disruptive cleanup to comply with coding standards
   - [ ] High-risk or disruptive change (requires upgrade path, risks regression, etc.)
- **Other or unknown**
   - [ ] Other or unknown

### Drupal core
- **Patch release changes**
   - [ ] Security update
   - [ ] Patch level release (non-security bug-fix release)
   - [ ] Patch removal that's no longer necessary
- **Minor release changes**
   - [ ] Major or minor level update
- **Other or unknown**
   - [ ] Other or unknown

### Drupal contrib projects
- **Patch release changes**
   - [ ] Security update
   - [ ] Patch or minor level update
   - [ ] Add new module
   - [ ] Patch removal that's no longer necessary
- **Minor release changes**
   - [ ] Major level update
- **Other or unknown**
   - [ ] Other or unknown

## Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: themes/custom/az_barrio/templates/forms/fieldset.html.twig [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    'form-item',
+    'js-form-wrapper',
+    'form-wrapper',
+    'mb-3',
+    errors ? 'has-error',
+  ]
+%}
+{%
+  set legend_classes = [
+    'col-form-label',
```
Comment (trackleft) [inline themes/custom/az_barrio/templates/forms/fieldset.html.twig:35]:
```suggestion
    'col-form-label',
    'font-weight-bold',
```

File: modules/custom/az_core/src/Plugin/views/exposed_form/QuickstartExposedFilters.php [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  /**
+   * {@inheritdoc}
+   */
+  public function exposedFormAlter(&$form, FormStateInterface $form_state) {
+    parent::exposedFormAlter($form, $form_state);
+
+    // Attach Quickstart styles.
+    $form['#attached']['library'][] = 'az_core/az-bef-sidebar';
+    // Vertical style intended for sidebar use.
+    $form['#attributes']['class'][] = 'az-bef-vertical';
```
Comment (danahertzberg) [inline modules/custom/az_core/src/Plugin/views/exposed_form/QuickstartExposedFilters.php:30]: Looks like this isn't working as expected. I used the Quickstart Exposed Form filter style on a view display and am not seeing the `az-bef-vertical` class applied in the inspector.

Comment (trackleft) [reply modules/custom/az_core/src/Plugin/views/exposed_form/QuickstartExposedFilters.php:30]: Seems to work now: https://az_quickstart.lndo.site/pages/news-view-paragraphs

**Comment by trackleft** [comment]
image

File: themes/custom/az_barrio/templates/blocks/block--exposedformaz-page-by-categoryrow-with-filters.html.twig [hunk 3]
```diff
@@ -0,0 +1,4 @@
+{% block content %}
```
Comment (trackleft) [inline themes/custom/az_barrio/templates/blocks/block--exposedformaz-page-by-categoryrow-with-filters.html.twig:1]: Is this even in use?  I have a feeling we can delete this file.

**Comment by danahertzberg** [comment]
This PR seems to change a lot more than the title indicates. Should we keep the tasks separate? I've been working on https://github.com/az-digital/az_quickstart/pull/2998 which addresses the BEF styles and demo content.

**Comment by trackleft** [comment]
I've updated the Issue summary so that it reflects the existing state of this PR (Now just one template file added and nothing else)

**Review by joeparsons — DISMISSED** [review]

**Review by danahertzberg — APPROVED** [review]

File: themes/custom/az_barrio/templates/forms/fieldset.html.twig [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    'font-weight-bold',
```
Comment (trackleft) [inline themes/custom/az_barrio/templates/forms/fieldset.html.twig:36]:
```suggestion
    'col-form-label',
    'font-weight-bold',
    required ? 'js-form-required',
    required ? 'form-required',


```

**Review by mmunro-ltrr — APPROVED** [review]
Checked that the class wasn't something Bootstrap 5 has deprecated (so no refactoring required for an eventual v5 upgrade).
https://getbootstrap.com/docs/5.2/forms/layout/

**Review by tadean — APPROVED** [review]

**Pull request merged**